test/lazy_map/core_test.clj

```diff
@@ -404,3 +404,34 @@
     (is (realized-at? m2 :a))
     (is (not (realized-at? m1 :b)))
     (is (not (contains? m2 :b)))))
+
+(deftest lazy-maps-support-metadata
+  (testing "metadata is orthogonal to value equality"
+    (let [a (literal->lazy-map {:test (+ 1 2)})
+          b (with-meta (literal->lazy-map {:test (+ 1 2)}) {:meta true})]
+      (is (= a b))))
+
+  (testing "metadata preservation matches other clojure map types"
+    (let [m  (with-meta {:x 1} {:test 1})
+          lm (with-meta (lazy-map {:x 1}) {:test 1})]
+      (is (= (meta (assoc m :y 2)) (meta (assoc lm :y 2))))
+      (is (= (meta (dissoc m :y 2)) (meta (dissoc lm :y 2))))
+      (is (= (meta (clojure.core/merge m {:x 3}))
+             (meta (lazy-map.core/merge lm {:x 3}))))))
+
+  (testing "literal metadata is supported"
+    (is (= {:test 1} (meta (literal->lazy-map ^{:test 1} {:thing 2})))))
+
+  (testing "metadata is supported"
+    (let [m (with-meta (literal->lazy-map {}) {:meta :data})]
+      (is (= {:meta :data} (meta m))))))
+
+(deftest function-edge-cases-match-other-functions
+  (testing "merging 0 elements returns nil"
+    (is (nil? (merge))))
+  (testing "merging 1 element returns element"
+    (is (= {:x 1} (merge {:x 1}))))
```

Comment:
We should test that metadata is preserved here. (although I can see from the code that it just returns its parameter)

Reply:
that is tested here: https://github.com/Guaranteed-Rate/lazy-map/blob/master/test/lazy_map/core_test.clj#L419-L420